### PR TITLE
Syntax: Add syntax based folding support

### PR DESCRIPTION
**Pull request by deathaxe**
This PR adds support for the following tmPreferences keys

- indentationFoldingEnabled
- scopeFoldingEnabled
- foldScopes

Some effort has been made to avoid highlighting the whole file invalid in various incomplete code scenarios.

### CODE REVIEW

File: Package/TextMate Preferences/Completions/Settings Keys.sublime-completions [hunk 1]
```diff
@@ -77,6 +77,21 @@
 			"contents": "<key>preserveIndent</key>\n<string>$1</string>",
 			"kind": ["snippet", "s", "setting"],
 		},
+		{
+			"trigger": "indentationFoldingEnabled",
+			"contents": "<key>indentationFoldingEnabled</key>\n<false/>",
```
Comment (Ultra-Instinct-05) [inline Package/TextMate Preferences/Completions/Settings Keys.sublime-completions:82]: As per what was told in Discord, `indentationFoldingEnabled` is `<true />` by default. Hence, I guess it doesn't make sense for the snippet to put a `<false />` as completion.

File: Package/TextMate Preferences/Completions/Settings Keys.sublime-completions [hunk 2]
```diff
@@ -77,6 +77,21 @@
...
+			"kind": ["snippet", "s", "setting"],
+		},
+		{
+			"trigger": "scopeFoldingEnabled",
+			"contents": "<key>scopeFoldingEnabled</key>\n<false/>",
```
Comment (Ultra-Instinct-05) [inline Package/TextMate Preferences/Completions/Settings Keys.sublime-completions:87]: Same as previous.

File: Package/TextMate Preferences/Completions/Settings Keys.sublime-completions (hunk 1 above)
Comment (Ultra-Instinct-05) [reply Package/TextMate Preferences/Completions/Settings Keys.sublime-completions:82]: Reason why I feel `true` makes more sense is:
- It's the default value. So having it set to `<true />` or not having the entire `<indentationFoldingEnabled><true />` is the same and hence causes the least surprise.
- The user may not know that `<true />` is the default.

Comment (jrappen) [reply Package/TextMate Preferences/Completions/Settings Keys.sublime-completions:82]: Make it `... <${1:true}/ >`, which should make (1) it clear that true is the default and (2) editing easy.

**Comment by deathaxe** [comment]
A full rework of Property Lists and tmPreferences is in progress.

**Pull request closed without merging**